src/soc/ibm/power9/chip.c

```diff
@@ -488,6 +490,42 @@ static void add_cb_fdt_data(struct device_tree *tree)
 	dt_add_reg_prop(coreboot_node, reg_addrs, reg_sizes, 2, addr_cells, size_cells);
 }
 
+static void add_tpm_node(struct device_tree *tree)
+{
+#if CONFIG(TALOS_2_INFINEON_TPM_1)
```

Comment:
This should depend on whether TPM is really selected in the menu or TPM is detected.
Furthermore I would do:

```
if (!CONFIG(SOEMTHING))
    return;
```

Reply:
> This should depend on whether TPM is really selected in the menu or TPM is detected.

If it wasn't selected, the code for detecting TPM wasn't compiled.

> Furthermore I would do:

Then it won't link when `TALOS_2_INFINEON_TPM_1` is off.

Reply:
It would link if it were inside this function. I think what miczyg said was to use normal C `if` instead of precompiler's `#if`, and reverse the condition checked.

Reply:
It can't link as is, because `tis_name` is not defined anywhere then, but I guess TPM code can be linked and unused. Might need to add a way to check whether TPM is present, it seems to be static at the moment. Or can leave it as is because skiboot disables TPM in DT if there is an issue with it and the next PR will remove
```c
dt_add_string_prop(tpm, "status", "okay");
```

Reply:
Right, I missed it.

TPM presence is static because it is often used together with measured/verified boot. Having it detectable at run time would make attacks as simple as disconnecting the TPM.